modules/core/src/utils/typed-array-manager.js

```diff
@@ -0,0 +1,66 @@
+export class TypedArrayManager {
+  constructor({overAlloc = 1, poolSize = 5} = {}) {
+    this.overAlloc = overAlloc;
+    this.poolSize = poolSize;
+
+    this._pool = {};
+  }
+
+  allocate(typedArray, count, {size, type, padding = 0, copy = false}) {
+    const newSize = count * size + padding;
+    if (typedArray && newSize <= typedArray.length) {
+      return typedArray;
+    }
+
+    // Allocate at least one element to ensure a valid buffer
+    const allocSize = Math.max(Math.ceil(newSize * this.overAlloc), 1);
+    const newArray = this._allocate(type, allocSize);
+
+    if (typedArray && copy) {
+      newArray.set(typedArray);
+    }
+
+    this._release(typedArray);
+    return newArray;
+  }
+
+  release(typedArray) {
+    this._release(typedArray);
+  }
+
+  _allocate(Type = Float32Array, size) {
+    // Check if available in pool
+    const typeName = Type.name;
+    const pool = this._pool[typeName];
+    if (pool) {
+      const i = pool.findIndex(array => array.length >= size);
+      if (i >= 0) {
+        return pool.splice(i, 1)[0];
```

Comment:
This linear search and splicing could become costly if the pool gets big. We should think about a better data structure.

Also, I'd make it `pool.splice(i, 1)[0].subarray(0, size);` so the length of the returned array is as expected.

Reply:
With a few hundred items the search is extremely cheap. I'll add a comment for future consideration.

Reply:
I had an idea for this. Make it that all arrays returned have byte lengths that are powers of 2. Each element in the pool would be an array of `ArrayBuffer`s of length `2 ** index`. Then the look up would just be 
```js
const index = Math.ceil(Math.log2(requestedByteLength));
if (pool[index] && pool[index].length > 0) {
    return new Type(pool[index].pop()).subarray(0, size);
} 
return new Type(Math.pow(2, index) / Type.BYTES_PER_ELEMENT).subarray(0, size);
```